README.md

```diff
@@ -0,0 +1,122 @@
+Common clang is a thin wrapper library around clang. Common clang has
+OpenCL-oriented API and is capable to compile OpenCL C kernels to SPIR-V
+modules.
+
+## Build
+
+Source code in this repo can be built in different manners:
+* in-tree as an LLVM project
+* out-of-tree using pre-built LLVM
+
+### In-tree build
+
+Before the build all dependencies must be downloaded and layed out as the
+following:
+
+```
+<workspace>
+`-- llvm
+    |-- tools
+    |   `-- clang
+    `-- projects
+        |-- llvm-spirv
+        `-- common-clang
+```
+
+This can be done using the following commands:
+```
+cd <workspace>
+git clone https://github.com/llvm-mirror/llvm.git -b release_70
+cd tools
+git clone https://github.com/llvm-mirror/clang.git -b release_70
+cd <workspace>/llvm/projects
+git clone https://github.com/KhronosGroup/SPIRV-LLVM-Translator.git -b llvm_release_70 llvm-spirv
+git clone https://github.com/intel/opencl-clang.git -b ocl-open-70
+```
+
+Then we need to create a build directory and run the build:
+```
+cd <workspace>
+mkdir build && cd build
+cmake -DLLVM_TARGETS_TO_BUILD="X86" ../llvm
+make common_clang -j`nproc`
+```
+
+For sanity check of the built please run `make check-clang` and
+`make check-llvm-spirv`
+
+### Out-of-tree build
+
+To build Common clang as standalone project, you need to obtain pre-built LLVM
+and SPIR-V Translator libraries. **Note:** currently this kind of build is
+supported on Linux only.
+
+Integration with pre-built LLVM is done using standard `find_package` way as
+documented in [Embedding LLVM in your project](https://llvm.org/docs/CMake.html#embedding-llvm-in-your-project).
+
+Commands to checkout sources and build:
+```
+cd <workspace>
+git clone https://github.com/intel/opencl-clang.git -b ocl-open-70
+mkdir build && cd build
+cmake ../opencl-clang
+make all -j`nproc`
+```
+
+#### Configuration options
+
+##### Preferred LLVM version
+
+By default, Common clang's cmake script is searching for LLVM 7.0.0. You can
```

Comment:
On the master branch we should search for the latest LLVM.

Reply:
Done